### PR TITLE
feat: LLM profiles UI components, modals, and tests (PR B)

### PR DESCRIPTION
**Pull request by malhotra5**
## Summary

This is **Part B** of the LLM profiles feature split (from #210).

**Depends on**: #387 (PR A - API layer + hooks) ✅ APPROVED

### What's included

**Components** (~1,600 lines):
- `LlmProfilesManager` - Main container managing list/edit views
- `ProfileRow` - Display individual profiles with active state indicator
- `ProfileActionsMenu` - Dropdown menus for profile actions (Edit, Rename, Set Active, Delete)
- `ProfileNameInput` - Editable name field with validation
- `ProfilesBody` - Container for profile list content with loading/error states
- `RenameProfileModal` - Modal for renaming profiles
- `DeleteProfileModal` - Confirmation modal for deletion

**Supporting components**:
- `api-key-modal-base.tsx` - Base modal component with focus trap and accessibility
- `brand-button.tsx` - Styled button component with tertiary variant added

**i18n**: All profile-related translation keys merged with main branch translations

**Tests**: 104 component tests covering all new components including:
- Keyboard navigation tests for menu
- Boundary condition tests for name validation
- Loading/pending state tests

#### Profile List with Active Badge
Profile List

#### Profile Actions Menu
Profile Actions Menu

#### Rename Profile Modal
Rename Profile Modal

#### Delete Profile Modal
Delete Profile Modal

### Key improvements in this revision

- **Styling**: Active badge uses green success color, Add Profile button uses secondary variant, Cancel buttons use tertiary (gray) variant
- **Accessibility**: Added aria-labelledby, aria-modal, role=dialog to modals
- **Code quality**: Extracted MenuItem component to reduce duplication
- **UX**: Added accessible loading state with sr-only text and aria-busy
- **Validation**: Trim whitespace on input change for cleaner UX
- **Debugging**: Added console.error for failed profile activations

### Files changed
- `src/components/features/settings/llm-profiles/` (8 files)
- `src/components/features/settings/` (2 supporting files)
- `__tests__/components/settings/llm-profiles/` (8 test files)
- `src/i18n/translation.json` (merged with profile keys)

### What's NOT included (next PRs)
- Route integration (`/settings/llm`)
- LlmSettingsLocalView integration
- Final wiring to settings navigation

### Testing
- [x] `npm run typecheck` passes
- [x] `npm run build` passes
- [x] All 104 component tests pass

---
*This PR was created by an AI agent (OpenHands) as part of splitting #210 into smaller incremental PRs.*

### CODE REVIEW

**Comment by malhotra5** [comment]
Addressed UI refactoring in commit 1aef619. Updated components to match the reference implementation:

- ProfileRow now accepts `isActive`, `onActivate`, `isActivating` props
- ProfileActionsMenu includes "Set as active" option
- Active badge uses primary color (gold background, dark text) matching the OpenHands reference
- Removed duplicate components (ProfileListRow, ProfileListActionsMenu, LlmProfilesListView)
- Updated all tests to match new component interfaces

Ready for another review.

File: src/components/features/settings/api-key-modal-base.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+  return (
+    <ModalBackdrop onClose={onClose} aria-label={title}>
+      <div
+        ref={modalRef}
+        tabIndex={-1}
+        className="bg-base-secondary p-6 rounded-xl flex flex-col gap-4 border border-tertiary"
+        style={{ width }}
+      >
+        <h3 className="text-xl font-bold">{title}</h3>
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply src/components/features/settings/api-key-modal-base.tsx]: Fixed in 22d4584 - Added role="dialog", aria-modal="true", and aria-labelledby="modal-title" with corresponding id on the h3 element.

File: src/components/features/settings/llm-profiles/profile-actions-menu.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        }}
+        type="button"
+        onClick={() => handleAction(onDelete)}
+        onKeyDown={(e) => handleKeyDown(e, 3)}
+        className="w-full text-left px-4 py-2 text-sm text-red-400 hover:bg-tertiary cursor-pointer"
+        role="menuitem"
+        data-testid="profile-delete"
+      >
+        {t(I18nKey.BUTTON$DELETE)}
+      </button>
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply src/components/features/settings/llm-profiles/profile-actions-menu.tsx]: Fixed in 22d4584 - Extracted MenuItem component to reduce code duplication from 50+ lines to ~20 lines.

File: src/components/features/settings/llm-profiles/delete-profile-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        type="button"
+        variant="danger"
+        className="grow"
+        onClick={handleDelete}
+        isDisabled={deleteProfile.isPending}
+      >
+        {deleteProfile.isPending ? (
+          <LoadingSpinner size="small" />
+        ) : (
+          t(I18nKey.BUTTON$DELETE)
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply src/components/features/settings/llm-profiles/delete-profile-modal.tsx:67]: Fixed in 22d4584 - Added sr-only text and aria-busy for accessible loading state.

File: src/components/features/settings/llm-profiles/rename-profile-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+      await renameProfile.mutateAsync({ name: profile.name, newName: trimmed });
+      displaySuccessToast(
+        t(I18nKey.SETTINGS$PROFILE_RENAMED, { name: trimmed }),
+      );
+      onClose();
+    } catch (error) {
+      const message =
+        error instanceof Error ? error.message : t(I18nKey.ERROR$GENERIC);
+      displayErrorToast(message);
+    }
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply src/components/features/settings/llm-profiles/rename-profile-modal.tsx:65]: Declined - This is a scope creep suggestion for a common pattern helper. The current 5-line try/catch is readable and follows project conventions. Creating a useMutationWithToast hook would be overkill for two occurrences.

File: src/components/features/settings/llm-profiles/rename-profile-modal.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+  const renameProfile = useRenameLlmProfile();
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    setNewName(profile?.name ?? "");
+  }, [profile?.name]);
+
+  if (!profile) return null;
+
+  const trimmed = newName.trim();
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply src/components/features/settings/llm-profiles/rename-profile-modal.tsx]: Fixed in 22d4584 - Now trimming whitespace on input change to avoid user confusion.

File: src/components/features/settings/llm-profiles/profiles-body.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+interface ProfilesBodyProps {
+  isLoading: boolean;
+  loadError: Error | null;
+  profiles: ProfileInfo[];
+  active: string | null;
+  onActivate: (name: string) => void;
+  onEdit: (profile: ProfileInfo) => void;
+  onRename: (profile: ProfileInfo) => void;
+  onDelete: (profile: ProfileInfo) => void;
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply src/components/features/settings/llm-profiles/profiles-body.tsx:15]: Declined - Prop drilling is acceptable for this small component hierarchy. Adding React Context would be over-engineering for 5 props across 3 components. The hierarchy is simple: Manager -> Body -> Row -> ActionsMenu.

File: src/components/features/settings/brand-button.tsx [hunk 7]
```diff
@@ -11,24 +12,31 @@ interface BrandButtonProps {
   startContent?: React.ReactNode;
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply src/components/features/settings/brand-button.tsx:12]: Fixed in 22d4584 - Added ariaLabel prop for accessible icon-only buttons.

File: __tests__/components/settings/llm-profiles/profile-actions-menu.test.tsx [hunk 8]
```diff
@@ -0,0 +1,193 @@
+import { render, screen, fireEvent } from "@testing-library/react";
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply __tests__/components/settings/llm-profiles/profile-actions-menu.test.tsx:1]: Fixed in 22d4584 - Added comprehensive keyboard navigation tests for ArrowUp/Down, Tab-to-close, auto-focus first item, and wrap-around focus.

File: __tests__/components/settings/llm-profiles/delete-profile-modal.test.tsx [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+    const handleClose = vi.fn();
+    renderModal(mockProfile, handleClose);
+
+    await user.click(screen.getByText("Cancel"));
+
+    expect(handleClose).toHaveBeenCalledTimes(1);
+    expect(ProfilesService.deleteProfile).not.toHaveBeenCalled();
+  });
+
+  it("calls deleteProfile and shows success toast on successful delete", async () => {
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply __tests__/components/settings/llm-profiles/delete-profile-modal.test.tsx:99]: Fixed in 22d4584 - Added 3 isPending state tests: prevents closing during deletion, disables Cancel button, disables Delete button.

File: __tests__/components/settings/llm-profiles/profile-name-input.test.tsx [hunk 10]
```diff
@@ -0,0 +1,197 @@
+import { render, screen } from "@testing-library/react";
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply __tests__/components/settings/llm-profiles/profile-name-input.test.tsx:1]: Fixed in 22d4584 - Added 7 boundary condition tests covering 64/65 char limits, numbers at start, special chars, and invalid starting characters.

File: src/components/features/settings/llm-profiles/profile-name-input.tsx [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+  ruleTestId?: string;
+  value: string;
+  onChange: (value: string) => void;
+  onKeyDown?: (e: React.KeyboardEvent<HTMLInputElement>) => void;
+  placeholder?: string;
+  isDisabled?: boolean;
+  /** Render label as "Name (Optional)" when this field isn't required. */
+  isOptional?: boolean;
+  /** When true, empty values will show red validation styling (required field behavior). */
+  isRequired?: boolean;
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply src/components/features/settings/llm-profiles/profile-name-input.tsx:18]: Declined - This is a minor suggestion that doesn't cause real issues. The props serve clear purposes and having both set would be a developer error, not a runtime issue. Can address in a follow-up if needed.

File: src/components/features/settings/llm-profiles/llm-profiles-manager.tsx [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+  const profiles = data?.profiles ?? [];
+  const active = data?.active_profile ?? null;
+
+  const handleActivate = async (name: string) => {
+    try {
+      await activateProfile.mutateAsync(name);
+      displaySuccessToast(t(I18nKey.SETTINGS$PROFILE_ACTIVATED, { name }));
+    } catch {
+      displayErrorToast(t(I18nKey.ERROR$GENERIC));
+    }
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply src/components/features/settings/llm-profiles/llm-profiles-manager.tsx:45]: Fixed in 22d4584 - Added console.error for debugging failed profile activations.

**Comment by malhotra5** [comment]
Addressed all review feedback in 22d4584. Ready for another review.
